pkg/api/utils_test.go

```diff
@@ -109,31 +109,57 @@ var tenantTests = map[string]struct {
 	spec       v1alpha1.TenantSpec
 }{
 	"empty": {
-		TenantProperties{},
-		v1alpha1.TenantSpec{},
```

Comment:
Because an empty spec is actually not valid. We always checked that a Tenant has a gitrepourl before calling `NewCRDFromAPITenant`, so I moved the check inside the function. This sadly makes the test a little more convoluted, as before we tested a simpler case (that actually never happened).